packages/flutter_tools/lib/src/commands/drive.dart

```diff
@@ -166,3 +171,155 @@ class DriveCommand extends RunCommand {
     return '${pathWithNoExtension}_test${path.extension(appFile)}';
   }
 }
+
+/// Finds a device to test on. May launch a simulator, if necessary.
+typedef Future<Device> TargetDeviceFinder();
+TargetDeviceFinder targetDeviceFinder = findTargetDevice;
+void restoreTargetDeviceFinder() {
+  targetDeviceFinder = findTargetDevice;
+}
+
+Future<Device> findTargetDevice() async {
```

Comment:
I think you want to update this logic to first check to see if the user has passed in an explicit device. That would be `deviceManager.hasSpecifiedDeviceId` and `deviceManager.getDeviceById()`.